### PR TITLE
Disable PM popup via setting

### PR DESCRIPTION
**Pull request by devo1929**

### CODE REVIEW

**Comment by alexp8** [comment]
Tested with @devo1929 and works great well done :)

cc @Metadorius

File: DXMainClient/Online/PrivateMessageHandler.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+        private void _PrivateMessageReceived(object sender, CnCNetPrivateMessageEventArgs e)
+        {
+            IRCUser iu = _connectionManager.UserList.Find(u => u.Name == e.Sender);
+
+            // We don't accept PMs from people who we don't share any channels with
+            if (iu == null)
+            {
+                return;
+            }
```
Comment (Metadorius) [inline DXMainClient/Online/PrivateMessageHandler.cs:40]:
```suggestion
            if (iu == null)
                return;
```

File: DXMainClient/Online/PrivateMessageHandler.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+            // Messages from users we've blocked are not wanted
+            if (_cncnetUserData.IsIgnored(iu.Ident))
+            {
+                return;
+            }
```
Comment (Metadorius) [inline DXMainClient/Online/PrivateMessageHandler.cs:46]:
```suggestion
            if (_cncnetUserData.IsIgnored(iu.Ident))
                return;
```

File: DXMainClient/Online/PrivateMessageHandler.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+            var privateMessageEventArgs = new PrivateMessageEventArgs(e.Sender, e.Message, iu);
+
+            PrivateMessageReceived?.Invoke(this, privateMessageEventArgs);
+        }
+
+        private void DoUnreadMessageCountUpdated()
+        {
+            UnreadMessageCountUpdated?.Invoke(this, new UnreadMessageCountEventArgs(UnreadMessageCount));
+        }
```
Comment (Metadorius) [inline DXMainClient/Online/PrivateMessageHandler.cs:56]:
```suggestion
        private void DoUnreadMessageCountUpdated()
            => UnreadMessageCountUpdated?.Invoke(this, new UnreadMessageCountEventArgs(UnreadMessageCount));
```

File: DXMainClient/Online/PrivateMessageHandler.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+        /// <summary>
+        /// This can be called by specific GUI components to trigger than any unread counts should be reset,
+        /// because the PrivateMessageWindow was made visible.
+        /// </summary>
+        public void ResetUnreadMessageCount()
+        {
+            SetUnreadMessageCount(0);
+        }
```
Comment (Metadorius) [inline DXMainClient/Online/PrivateMessageHandler.cs:71]:
```suggestion
        public void ResetUnreadMessageCount()
            => SetUnreadMessageCount(0);
```

File: DXMainClient/Online/PrivateMessageHandler.cs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+        /// <summary>
+        /// This can be called by specific GUI components to trigger than any unread counts should be incremented,
+        /// because the PrivateMessageWindow may not currently be visible.
+        /// </summary>
+        public void IncrementUnreadMessageCount()
+        {
+            SetUnreadMessageCount(UnreadMessageCount + 1);
+        }
```
Comment (Metadorius) [inline DXMainClient/Online/PrivateMessageHandler.cs:80]:
```suggestion
        public void IncrementUnreadMessageCount()
            => SetUnreadMessageCount(UnreadMessageCount + 1);
```

File: DXMainClient/DXGUI/Generic/TopBar.cs [hunk 6]
```diff
@@ -213,6 +222,21 @@ public override void Initialize()
...
+        }
+
+        private void UpdatePrivateMessagesBtnLabel(int unreadMessageCount)
+        {
+            btnPrivateMessages.Text = DEFAULT_PM_BTN_LABEL;
+            if (unreadMessageCount > 0)
+            {
+                // btnPrivateMessages.Text += $" ({unreadMessageCount})";
+            }
         }
```
Comment (Metadorius) [inline DXMainClient/DXGUI/Generic/TopBar.cs:240]: Any reason to leave this in such state?

File: DXMainClient/DXGUI/Generic/TopBar.cs [hunk 7]
```diff
@@ -213,6 +222,21 @@ public override void Initialize()
             connectionManager.AttemptedServerChanged += ConnectionManager_AttemptedServerChanged;
             connectionManager.ConnectAttemptFailed += ConnectionManager_ConnectAttemptFailed;
 
+            privateMessageHandler.UnreadMessageCountUpdated += PrivateMessageHandler_UnreadMessageCountUpdated;
+        }
+
+        private void PrivateMessageHandler_UnreadMessageCountUpdated(object sender, UnreadMessageCountEventArgs args)
+        {
+            UpdatePrivateMessagesBtnLabel(args.UnreadMessageCount);
+        }
```
Comment (Metadorius) [inline DXMainClient/DXGUI/Generic/TopBar.cs:231]:
```suggestion
        private void PrivateMessageHandler_UnreadMessageCountUpdated(object sender, UnreadMessageCountEventArgs args)
            => UpdatePrivateMessagesBtnLabel(args.UnreadMessageCount);
```

**Review by Metadorius — CHANGES_REQUESTED** [review]
The code is OK, I left a bit of style notes in the comments. We use expression bodies for one-line statements in the client code.

**Comment by devo1929** [comment]
@Metadorius I addressed all of your suggestions.

**Review by Metadorius — APPROVED** [review]

**Pull request merged**